compiler/resolution/wrappers.cpp

```diff
@@ -2017,14 +2017,70 @@ static Symbol* leadingArg(PromotionInfo& promotion, CallExpr* call) {
   return NULL;
 }
 
+// Returns true if we do not know how to handle this pattern.
+// Handles a corner case of a "nested call", for example "A.foo()"
+// where A.foo is a promoted call to a paren-less function foo. Ex.
+//   test/functions/promotion/issue-12736-ok.chpl
+//
+// If so, 'irTemp' in addSetIteratorShape() is the IR (iterator record)
+// for the outer call and we do not yet have the IR for the inner call.
+// So we cannot "set shape" on this inner IR. This is most likely OK
+// because the inner IR feeds directly into the outer call. The outer call
+// should also be promoted and inherit the shape from the inner call.
+//
+// We will go ahead and set the shape for 'irTemp' right away from the shape
+// of the inner call. We will also remember 'irTemp' to avoid setting its
+// shape the second time, which would not work. See dontSetIteratorShape2().
+//
+static bool dontSetIteratorShape1(CallExpr* &move, CallExpr* call,
```

Comment:
Can we rename this to `adjustMoveForSetIteratorShape` as I believe that's the most important thing it is doing. Then the fact that it returns bool can be confusing, if you think that way, you can have an argument `bool &settable` or something.